interface.go

```diff
@@ -211,6 +211,18 @@ type EarlyConnection interface {
 	NextConnection() Connection
 }
 
+// A ConnectionIDGenerator is a interface that allows clients to implement their own format
+// for the Connection IDs that servers/clients use as SrcConnectionID in QUIC packets.
+type ConnectionIDGenerator interface {
+
+	// GenerateConnectionID generates a new ConnectionID.
+	GenerateConnectionID() ([]byte, error)
```

Comment:
Is there any use case where you'd want to know which connection you're generating the Connection ID for? With the current interface you only know that you're generating a CID for _any_ connection the server handling.

Reply:
Initially I taught about extending the API to allow that. But I don't have a use-case for that now, and doing so it would just open the door for creating an implementation that would potentially introduce the correlation of connection IDs that shouldn't exist.

that's why I kept it like that.